x-pack/plugins/security_solution/public/timelines/components/timeline/unified_components/index.test.tsx

```diff
@@ -0,0 +1,629 @@
+        expect(mockTimelineFilterManager.addFilters).toHaveBeenNthCalledWith(
+          1,
+          expect.arrayContaining([
+            expect.objectContaining({
+              query: {
+                exists: {
+                  field: field.name,
+                },
+              },
+              meta: expect.objectContaining({
+                negate: false,
+                disabled: false,
+              }),
+            }),
+          ])
+        );
+      });
+    });
+    it('should remove the column when clicked on ⊕ sign', async () => {
+      const field = {
+        name: 'agent.id',
+      };
+
+      renderTestComponents();
+      expect(await screen.findByTestId('timeline-sidebar')).toBeVisible();
+
+      await waitFor(() => {
+        expect(screen.getByTestId('fieldListGroupedSelectedFields-count')).toBeVisible();
+      });
+
+      expect(screen.getByTestId('fieldListGroupedSelectedFields-count')).toHaveTextContent('9');
+
+      expect(screen.queryAllByTestId(`dataGridHeaderCell-${field.name}`)).toHaveLength(0);
+
+      const availableFields = screen.getByTestId('fieldListGroupedAvailableFields');
+
+      // / new columns does not exists yet
+      const currentColumns = customStore.getState().timeline.timelineById[TimelineId.test].columns;
+      expect(currentColumns).not.toMatchObject(
+        expect.arrayContaining([
+          expect.objectContaining({
+            id: field.name,
+          }),
+        ])
+      );
+
+      fireEvent.click(within(availableFields).getByTestId(`fieldToggle-${field.name}`));
+
+      // new columns should exist now
+      await waitFor(() => {
+        const newColumns = customStore.getState().timeline.timelineById[TimelineId.test].columns;
+        expect(newColumns).toMatchObject(
+          expect.arrayContaining([
+            expect.objectContaining({
+              id: field.name,
+            }),
+          ])
+        );
+      });
+    });
+
+    it('should remove the column when clicked on X sign', async () => {
+      const field = {
+        name: 'event.severity',
+      };
+
+      renderTestComponents();
+      expect(await screen.findByTestId('timeline-sidebar')).toBeVisible();
+
+      await waitFor(() => {
+        expect(screen.getByTestId('fieldListGroupedAvailableFields-count')).toBeVisible();
+      });
+
+      expect(screen.queryAllByTestId(`dataGridHeaderCell-${field.name}`)).toHaveLength(1);
+
+      const availableFields = screen.getByTestId('fieldListGroupedAvailableFields');
+
+      // / new columns does exists
+      const currentColumns = customStore.getState().timeline.timelineById[TimelineId.test].columns;
+      expect(currentColumns).toMatchObject(
+        expect.arrayContaining([
+          expect.objectContaining({
+            id: field.name,
+          }),
+        ])
+      );
+
+      fireEvent.click(within(availableFields).getByTestId(`fieldToggle-${field.name}`));
+
+      // new columns should exist now
```

Comment:
Shouldn't this say the column doesn't exist now? If I'm reading correctly, this one tests removing an existing column?

Reply:
Corrected and made variable name also more clear : cf159a76c519125e688742d8e6fd973788f47d8d